packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.js

```diff
@@ -0,0 +1,141 @@
+        for (var dir of dirs) {
+            var base = path.join(dir, requestPath)
+            if (fileExt) {
+                const noExtPath = requestPath.replace(fileExt, '')
+                if (this.extendsHashMap.has(noExtPath)) {
+                    return base
+                }
+            } else {
+                if (this.extendsHashMap.has(requestPath)) {
+                    const end = this.extendsHashMap.get(requestPath)[1]
+
+                    if (end[0] === 'index') {
+                        base = path.join(base, this.extendsHashMap.get(requestPath)[1].join(''))
+                        return base
+                    } else {
+                        base = base?.replace(/$\//, '') + end.join('')
+                        return base
+                    }
+                }
+            }
+        }
+    }
+
+    toOverrideRelative(path) {
+        var override = this.findOverride(path)
+        return path.substring(override.length + 1)
+    }
+
+    findOverride(path) {
+        return this._allSearchDirs.find((override) => {
+            return path.indexOf(override) === 0
+        })
+    }
+
+    isFromExtends(request, path) {
+        return (
+            request &&
+            this.extends.includes(request?.split('/')?.[0]) &&
+            // this is very important, to avoid circular imports, check that the
+            // `issuer` (requesting context) isn't the overrides directory
+            !path.match(this.projectDir + this.overridesDir)
+        )
+    }
+
+    apply(resolver) {
+        resolver.getHook('resolve').tapAsync(
+            'FeatureResolverPlugin',
+            function (requestContext, resolveContext, callback) {
+                if (
+                    requestContext.request &&
+                    this.isFromExtends(requestContext.request, requestContext.path)
+                ) {
+                    let overrideRelative = this.toOverrideRelative(requestContext.request)?.replace(
+                        /$\//,
+                        ''
+                    )
+                    try {
+                        const targetFile = this.findFileMap(overrideRelative, this._allSearchDirs)
+                        if (targetFile) {
+                            const target = resolver.ensureHook('resolved')
+                            requestContext.path = targetFile
+                            resolver.doResolve(
+                                target,
+                                requestContext,
+                                `${this.constructor.name} found base override file`,
+                                resolveContext,
+                                callback
+                            )
+                        } else {
+                            return callback()
+                        }
+                    } catch (e) {
+                        return callback()
+                    }
+                } else {
+                    callback()
+                }
```

Comment:
Is there a way to organize the code so we aren't invoking the callback in 3 different places, maybe there is something we can invert?
Also, I noticed that you are returning the callback in 2 places but not in the third.

Reply:
Perhaps, I can give it some thought